numpy/distutils/system_info.py

```diff
@@ -729,69 +729,59 @@ def check_libs2(self, lib_dirs, libs, opt_libs=[]):
 
         return info
 
-    def _lib_list(self, lib_dir, libs, exts):
+    def _find_lib(self, lib_dir, lib, exts):
         assert is_string(lib_dir)
-        liblist = []
         # under windows first try without 'lib' prefix
         if sys.platform == 'win32':
             lib_prefixes = ['', 'lib']
         else:
             lib_prefixes = ['lib']
         # for each library name, see if we can find a file for it.
-        for l in libs:
-            for ext in exts:
-                for prefix in lib_prefixes:
-                    p = self.combine_paths(lib_dir, prefix + l + ext)
-                    if p:
-                        break
+        for ext in exts:
+            for prefix in lib_prefixes:
+                p = self.combine_paths(lib_dir, prefix + lib + ext)
                 if p:
-                    assert len(p) == 1
-                    # ??? splitext on p[0] would do this for cygwin
-                    # doesn't seem correct
-                    if ext == '.dll.a':
-                        l += '.dll'
-                    liblist.append(l)
                     break
```

Comment:
You break on `p`, then immediately after the loop, return on `p`. So why not just return in the first place?